hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/ProtobufRpcEngine2.java

```diff
@@ -425,28 +426,39 @@ static class ProtobufRpcEngineCallbackImpl
       private final RPC.Server server;
       private final Call call;
       private final String methodName;
-      private final long setupTime;
+      private final long callStartNanos;
```

Comment:
Can callStartNanos be made an attribute of the call object, so that thread variables are not needed?

Reply:
@KeeProMise Sir, Thanks a lot.  Have fixed.